### PR TITLE
Refactor of the neural sparse search tutorial

### PR DESCRIPTION
**Pull request by zhichao-aws**
### Description
Recently we get the feedback that it's hard to find using neural sparse search with raw sparse vectors in existing documentations. Besides, the neural sparse search tutorial is not well structured compared with neural search [tutorial](https://opensearch.org/docs/latest/search-plugins/neural-search-tutorial).

This PR refact the neural sparse search tutorial. Mainly for these points:
1. Using seperate sections neural sparse search with built-in pipelines and with raw sparse vectors
2. Add a seperate sub section about `Set up an ML sparse encoding model`. It takes a ref from https://opensearch.org/docs/latest/search-plugins/neural-search-tutorial/#step-1-set-up-an-ml-language-model, but with more contents about 2 working mode in neural sparse, and a table to show the model combinations we offer in OpenSearch (we'll release v2 models soon, and present the combinations in a table will be more clear)
3. Refact the sections/subsections structure to give a more clear view

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._

### Version
2.15, 2.16

### Frontend features
_If you're submitting documentation for an OpenSearch Dashboards feature, add a video that shows how a user will interact with the UI step by step. A voiceover is optional._ 

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by zhichao-aws** [comment]
This PR is ready for review, thanks!

File: _search-plugins/neural-sparse-with-pipelines.md [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+}
+```
+
+## Accelerating neural sparse search
+
+To learn more about improving retrieval time in neural sparse search, see [Accelerating neural sparse search]({{site.url}}{{site.baseurl}}/search-plugins/neural-sparse-search/#accelerating-neural-sparse-search).
+
+## Creating a search pipeline for neural sparse search
+
+You can create a search pipeline that augments neural sparse search functionality:
```
Comment (natebower) [inline _search-plugins/neural-sparse-with-pipelines.md:430]:
```suggestion
You can create a search pipeline that augments neural sparse search functionality by:
```

File: _search-plugins/neural-sparse-with-pipelines.md [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+
+- Accelerates neural sparse search for faster retrieval.
```
Comment (natebower) [inline _search-plugins/neural-sparse-with-pipelines.md:432]:
```suggestion
- Accelerating neural sparse search for faster retrieval.
```

File: _search-plugins/neural-sparse-with-pipelines.md [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+- Sets the default model ID on an index for easier use.
```
Comment (natebower) [inline _search-plugins/neural-sparse-with-pipelines.md:433]:
```suggestion
- Setting the default model ID on an index for easier use. 
```

File: _search-plugins/neural-sparse-with-raw-vectors.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+# Using raw vectors for neural sparse search
+
+If you're using self-hosted sparse embedding models, you can ingest raw sparse vectors and use neural sparse search. 
+
+## Tutorial
+
+This tutorial consists of the following steps:
+
+1. [**Ingest sparse vectors**](#step-1-ingest-sparse-vectors).
```
Comment (natebower) [inline _search-plugins/neural-sparse-with-raw-vectors.md:17]: No periods necessary in this list.

File: _search-plugins/neural-sparse-with-raw-vectors.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    1. [Create an index](#step-1a-create-an-index).
+    1. [Ingest documents into the index](#step-1b-ingest-documents-into-the-index).
+1. [**Search the data using raw sparse vector**](#step-2-search-the-data-using-a-sparse-vector).
```
Comment (natebower) [inline _search-plugins/neural-sparse-with-raw-vectors.md:20]:
```suggestion
1. [**Search the data using a sparse vector**](#step-2-search-the-data-using-a-sparse-vector).
```

File: _search-plugins/neural-sparse-with-raw-vectors.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+
+## Step 1: Ingest sparse vectors
+
+Once you have generated sparse vector embeddings, you can ingest them into OpenSearch directly.
```
Comment (natebower) [inline _search-plugins/neural-sparse-with-raw-vectors.md:25]: "directly ingest"?

File: _search-plugins/neural-sparse-with-raw-vectors.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+### Step 1(a): Create an index
+
+In order to ingest documents of raw sparse vectors, create a rank features index:
```
Comment (natebower) [inline _search-plugins/neural-sparse-with-raw-vectors.md:29]: Is "of" the right word here, or do we mean something like "containing" or "with"?

File: _search-plugins/neural-sparse-with-raw-vectors.md [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+## Accelerating neural sparse search
+
+To learn more about improving retrieval time in neural sparse search, see [Accelerating neural sparse search]({{site.url}}{{site.baseurl}}/search-plugins/neural-sparse-search/#accelerating-neural-sparse-search).
```
Comment (natebower) [inline _search-plugins/neural-sparse-with-raw-vectors.md:99]:
```suggestion
To learn more about improving retrieval time for neural sparse search, see [Accelerating neural sparse search]({{site.url}}{{site.baseurl}}/search-plugins/neural-sparse-search/#accelerating-neural-sparse-search).
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws @zhichao-aws Please see my comments and changes and let me know if you have any questions. Thanks!

File: _search-plugins/neural-sparse-search.md [hunk 9]
```diff
@@ -14,261 +14,20 @@ Introduced 2.11
...
+You can configure neural sparse search in the following ways:
 
-**PREREQUISITE**<br>
-Before using neural sparse search, make sure to set up a [pretrained sparse embedding model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/pretrained-models/#sparse-encoding-models) or your own sparse embedding model. For more information, see [Choosing a model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/integrating-ml-models/#choosing-a-model).
-{: .note}
+- Generate vector embeddings within OpenSearch: Configure an ingest pipeline to generate and store sparse vector embeddings from document text at ingestion time. At query time, input plain text, which will be automatically converted into vector embeddings for search. For complete setup steps, see [Configuring ingest pipelines for neural sparse search]({{site.url}}{{site.baseurl}}/search-plugins/neural-sparse-with-pipelines/).
+- Ingest raw sparse vectors and search using them directly. For complete setup steps, see [Ingesting and searching raw vectors]({{site.url}}{{site.baseurl}}/search-plugins/neural-sparse-with-raw-vectors/).
```
Comment (kolchfa-aws) [inline _search-plugins/neural-sparse-search.md:22]:
```suggestion
- Ingest raw sparse vectors and search using sparse vectors directly. For complete setup steps, see [Ingesting and searching raw vectors]({{site.url}}{{site.baseurl}}/search-plugins/neural-sparse-with-raw-vectors/).
```

[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _search-plugins/neural-sparse-search.md:22]: Reworded.

File: _search-plugins/neural-sparse-with-pipelines.md [hunk 10]
```diff
@@ -0,0 +1,486 @@
...
+has_children: false
+---
+
+# Configuring ingest pipelines for neural sparse search
+
+Generating sparse vector embeddings within OpenSearch enables neural sparse search to function like lexical search. To take advantage of this encapsulation, set up an ingest pipeline to create and store sparse vector embeddings from document text during ingestion. At query time, input plain text, which will be automatically converted into vector embeddings for search.
+
+For this tutorial, you’ll use neural sparse search with OpenSearch’s built-in ML model hosting and ingest pipelines. Because the transformation of text to embeddings is performed within OpenSearch, you'll use text when ingesting and searching documents. 
+
+At ingestion time, neural sparse search uses a sparse encoding to generate sparse vector embeddings from text fields during ingestion.
```
Comment (kolchfa-aws) [inline _search-plugins/neural-sparse-with-pipelines.md:15]:
```suggestion
At ingestion time, neural sparse search uses a sparse encoding model to generate sparse vector embeddings from text fields. 
```

File: _search-plugins/neural-sparse-with-pipelines.md [hunk 11]
```diff
@@ -0,0 +1,486 @@
...
+- **Doc-only mode** (requires a sparse encoding model and a tokenizer): A sparse encoding model generates sparse vector embeddings from query text. In this mode, neural sparse search tokenizes query text using a tokenizer and obtains the token weights from a lookup table. This approach provides faster retrieval at the cost of a slight decrease in search relevance. The tokenizer is deployed and invoked using the [Model API]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/model-apis/index/) for a uniform neural sparse search experience.
+
+For more information about choosing the neural sparse search mode that best suits your workload, see [Choose the search mode](#step-1a-choose-the-search-mode).
+
+## Tutorial
+
+This tutorial consists of the following steps:
+
+1. [**Configure a sparse encoding model/tokenizer**](#step-1-configure-a-sparse-encoding-modeltokenizer).
+    1. [Choose the search mode](#step-1a-choose-the-search-mode).
```
Comment (kolchfa-aws) [inline _search-plugins/neural-sparse-with-pipelines.md:30]:
```suggestion
    1. [Choose the search mode](#step-1a-choose-the-search-mode)
```

File: _search-plugins/neural-sparse-with-pipelines.md [hunk 12]
```diff
@@ -0,0 +1,486 @@
...
+    1. [Register the model/tokenizer](#step-1b-register-the-modeltokenizer).
```
Comment (kolchfa-aws) [inline _search-plugins/neural-sparse-with-pipelines.md:31]:
```suggestion
    1. [Register the model/tokenizer](#step-1b-register-the-modeltokenizer)
```

File: _search-plugins/neural-sparse-with-pipelines.md [hunk 13]
```diff
@@ -0,0 +1,486 @@
...
+    1. [Deploy the model/tokenizer](#step-1c-deploy-the-modeltokenizer).
```
Comment (kolchfa-aws) [inline _search-plugins/neural-sparse-with-pipelines.md:32]:
```suggestion
    1. [Deploy the model/tokenizer](#step-1c-deploy-the-modeltokenizer)
```

File: _search-plugins/neural-sparse-with-pipelines.md [hunk 14]
```diff
@@ -0,0 +1,486 @@
...
+1. [**Ingest data**](#step-2-ingest-data).
```
Comment (kolchfa-aws) [inline _search-plugins/neural-sparse-with-pipelines.md:33]:
```suggestion
1. [**Ingest data**](#step-2-ingest-data)
```

File: _search-plugins/neural-sparse-with-pipelines.md [hunk 15]
```diff
@@ -0,0 +1,486 @@
...
+    1. [Create an ingest pipeline](#step-2a-create-an-ingest-pipeline).
```
Comment (kolchfa-aws) [inline _search-plugins/neural-sparse-with-pipelines.md:34]:
```suggestion
    1. [Create an ingest pipeline](#step-2a-create-an-ingest-pipeline)
```

File: _search-plugins/neural-sparse-with-pipelines.md [hunk 16]
```diff
@@ -0,0 +1,486 @@
...
+    1. [Create an index for ingestion](#step-2b-create-an-index-for-ingestion).
+    1. [Ingest documents into the index](#step-2c-ingest-documents-into-the-index).
```
Comment (kolchfa-aws) [inline _search-plugins/neural-sparse-with-pipelines.md:36]:
```suggestion
    1. [Ingest documents into the index](#step-2c-ingest-documents-into-the-index)
```

File: _search-plugins/neural-sparse-with-pipelines.md [hunk 17]
```diff
@@ -0,0 +1,486 @@
...
+1. [**Search the data**](#step-3-search-the-data).
```
Comment (kolchfa-aws) [inline _search-plugins/neural-sparse-with-pipelines.md:37]:
```suggestion
1. [**Search the data**](#step-3-search-the-data)
```

File: _search-plugins/neural-sparse-with-pipelines.md [hunk 18]
```diff
@@ -0,0 +1,486 @@
...
+  ]
+}
+```
+{% include copy-curl.html %}
+
+To split long text into passages, use the `text_chunking` ingest processor before the `sparse_encoding` processor. For more information, see [Text chunking]({{site.url}}{{site.baseurl}}/search-plugins/text-chunking/).
+
+### Step 2(b): Create an index for ingestion
+
+In order to use the sparse encoding processor defined in your pipeline, create a rank features index, adding the pipeline created in the previous step as the default pipeline. Ensure that the fields defined in the `field_map` are mapped as correct types. Continuing with the example, the `passage_embedding` field must be mapped as [`rank_features`]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/rank/#rank-features). Similarly, the `passage_text` field should be mapped as `text`.
```
Comment (kolchfa-aws) [inline _search-plugins/neural-sparse-with-pipelines.md:243]:
```suggestion
In order to use the sparse encoding processor defined in your pipeline, create a rank features index, adding the pipeline created in the previous step as the default pipeline. Ensure that the fields defined in the `field_map` are mapped as correct types. Continuing with the example, the `passage_embedding` field must be mapped as [`rank_features`]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/rank/#rank-features). Similarly, the `passage_text` field must be mapped as `text`.
```

File: _search-plugins/neural-sparse-with-raw-vectors.md (hunk 4 above)
Comment (kolchfa-aws) [inline _search-plugins/neural-sparse-with-raw-vectors.md:17]:
```suggestion
1. [**Ingest sparse vectors**](#step-1-ingest-sparse-vectors)
```

File: _search-plugins/neural-sparse-with-raw-vectors.md [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+# Using raw vectors for neural sparse search
+
+If you're using self-hosted sparse embedding models, you can ingest raw sparse vectors and use neural sparse search. 
+
+## Tutorial
+
+This tutorial consists of the following steps:
+
+1. [**Ingest sparse vectors**](#step-1-ingest-sparse-vectors).
+    1. [Create an index](#step-1a-create-an-index).
```
Comment (kolchfa-aws) [inline _search-plugins/neural-sparse-with-raw-vectors.md:18]:
```suggestion
    1. [Create an index](#step-1a-create-an-index)
```

File: _search-plugins/neural-sparse-with-raw-vectors.md [hunk 20]
```diff
@@ -0,0 +1,99 @@
...
+    1. [Ingest documents into the index](#step-1b-ingest-documents-into-the-index).
```
Comment (kolchfa-aws) [inline _search-plugins/neural-sparse-with-raw-vectors.md:19]:
```suggestion
    1. [Ingest documents into the index](#step-1b-ingest-documents-into-the-index)
```

File: _search-plugins/neural-sparse-with-raw-vectors.md (hunk 5 above)
Comment (kolchfa-aws) [inline _search-plugins/neural-sparse-with-raw-vectors.md:20]:
```suggestion
1. [**Search the data using raw sparse vector**](#step-2-search-the-data-using-a-sparse-vector)
```

File: _search-plugins/neural-sparse-with-raw-vectors.md (hunk 6 above)
Comment (kolchfa-aws) [inline _search-plugins/neural-sparse-with-raw-vectors.md:25]:
```suggestion
Once you have generated sparse vector embeddings, you can directly ingest them into OpenSearch.
```

File: _search-plugins/neural-sparse-with-raw-vectors.md (hunk 7 above)
Comment (kolchfa-aws) [inline _search-plugins/neural-sparse-with-raw-vectors.md:29]:
```suggestion
In order to ingest documents containing raw sparse vectors, create a rank features index:
```

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @zhichao-aws!

**Pull request merged**